File: app/models/product.rb
class Product
  include Mongoid::Document

  field :name, type: String
  field :type, type: String
  field :length, type: Integer
  field :width, type: Integer
  field :height, type: Integer
  field :weight, type: Integer
  field :volume, type: Integer

  validates :name, :type, :length, :width, :height, :weight, presence: true
  validates_numericality_of :length, :width, :height, :weight, only_integer: true, greater_than: 0

  before_save do |document|
    document.volume = length * height * weight
  end
end

File: frontend/app/models/product.js
import Model from 'ember-data/model';
import attr from 'ember-data/attr';
// import { belongsTo, hasMany } from 'ember-data/relationships';
import EmberValidations from 'ember-validations';

export default Model.extend(EmberValidations, {
  name: attr('string'),
  type: attr('string'),
  length: attr('number'),
  width: attr('number'),
  height: attr('number'),
  weight: attr('number'),

  validations: {
    name: {
      presence: true
    },
    type: {
      presence: true
    },
    length: {
      presence: true,
      numericality: {
        onlyInteger: true,
        greaterThan: 0
      }
    },
    width: {
      presence: true,
      numericality: {
        onlyInteger: true,
        greaterThan: 0
      }
    },
    height: {
      presence: true,
      numericality: {
        onlyInteger: true,
        greaterThan: 0
      }
    },
    weight: {
      presence: true,
      numericality: {
        onlyInteger: true,
        greaterThan: 0
      }
    }
  }
});
